app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cache_date = {}
+
+    def inner(*args, **kwargs) -> any:
+        if args in cache_date:
+            print("Getting from cache")
+            for date in cache_date:
+                if date == args:
```

Comment:
no need to iterate through the dict, u can just call date by the key, like in return